integration/tests/access/access_circuit_breaker_test.go

```diff
@@ -0,0 +1,197 @@
+// 2. Try to send a transaction multiple times to observe the decrease in waiting time for a failed response.
+// 3. Connect the collection node to the network and wait for the circuit breaker restore time.
+// 4. Successfully send a transaction.
+func (s *AccessCircuitBreakerSuite) TestCircuitBreaker() {
+	ctx, cancel := context.WithCancel(s.ctx)
+	defer cancel()
+
+	// 1. Get the collection node
+	collectionContainer := s.net.ContainerByName("collection_1")
+
+	// 2. Get the Access Node container and client
+	accessContainer := s.net.ContainerByName(testnet.PrimaryAN)
+
+	// Check if access node was created with circuit breaker flags
+	require.True(s.T(), accessContainer.IsFlagSet("circuit-breaker-enabled"))
+	require.True(s.T(), accessContainer.IsFlagSet("circuit-breaker-restore-timeout"))
+	require.True(s.T(), accessContainer.IsFlagSet("circuit-breaker-max-failures"))
+
+	accessClient, err := accessContainer.TestnetClient()
+	assert.NoError(s.T(), err, "failed to get collection node client")
+
+	latestBlockID, err := accessClient.GetLatestBlockID(ctx)
+	require.NoError(s.T(), err)
+
+	// Create a new account to deploy Counter to
+	accountPrivateKey := lib.RandomPrivateKey()
+
+	accountKey := sdk.NewAccountKey().
+		FromPrivateKey(accountPrivateKey).
+		SetHashAlgo(sdkcrypto.SHA3_256).
+		SetWeight(sdk.AccountKeyWeightThreshold)
+
+	serviceAddress := sdk.Address(accessClient.Chain.ServiceAddress())
+
+	// Generate the account creation transaction
+	createAccountTx, err := templates.CreateAccount(
+		[]*sdk.AccountKey{accountKey},
+		[]templates.Contract{
+			{
+				Name:   lib.CounterContract.Name,
+				Source: lib.CounterContract.ToCadence(),
+			},
+		}, serviceAddress)
+	require.NoError(s.T(), err)
+
+	createAccountTx.
+		SetReferenceBlockID(sdk.Identifier(latestBlockID)).
+		SetProposalKey(serviceAddress, 0, accessClient.GetSeqNumber()).
+		SetPayer(serviceAddress).
+		SetGasLimit(9999)
+
+	// Sign the transaction
+	childCtx, cancel := context.WithTimeout(ctx, time.Second*10)
+	signedTx, err := accessClient.SignTransaction(createAccountTx)
+	require.NoError(s.T(), err)
+	cancel()
+
+	// 3. Disconnect the collection node from the network to activate the Circuit Breaker
+	err = collectionContainer.Disconnect()
+	require.NoError(s.T(), err, "failed to pause connection node")
+
+	// 4. Send a couple of transactions to test if the circuit breaker opens correctly
+	sendTransaction := func(ctx context.Context, tx *sdk.Transaction) (time.Duration, error) {
+		childCtx, cancel = context.WithTimeout(ctx, time.Second*10)
+		start := time.Now()
+		err := accessClient.SendTransaction(childCtx, tx)
+		duration := time.Since(start)
+		defer cancel()
+
+		return duration, err
+	}
+
+	// Try to send the transaction for the first time. It should wait at least the timeout time and return Unknown error
+	duration, err := sendTransaction(ctx, signedTx)
+	assert.Equal(s.T(), codes.Unknown, status.Code(err))
```

Comment:
does the node currently return `Unknown`? It should really be `Unavailable`. If this isn't new from your code, we can update in another PR.